src/login/logind-button.c

```diff
@@ -413,10 +532,36 @@ static int button_suitable(int fd) {
         return false;
 }
 
+static void button_get_modifier_mask(Button *b) {
+        unsigned long keys[KEY_MAX/ULONG_BITS+1];
+
+        if (ioctl(b->fd, EVIOCGKEY(sizeof(keys)), keys) >= 0) {
```

Comment:
we generally prefer shallow indentation. i.e. exit early rather then nest deep.

or in other words, please invert this check, and return quickly, so that the rest of the function doesn't need indentation.

also, please at least debug log here if this fails, i.e. log_debug_errno(errno, "…: %m");

Reply:
hmm, frankly, i'd propagate the error code here, too, i.e. not return void, but `int`. let the caller ignore the error if it wants.

Reply:
Yeah, I think I'll do the `log_warning_errno` in `button_open`  instead and make `button_get_modifier_mask`  `return -errno;` since that seems to be where the other `log_warning_errno`s are, and remove the indent